vowpalwabbit/cats_tree.cc

```diff
@@ -373,13 +368,19 @@ namespace VW {
       if (!options.was_supplied("cats_tree"))
         return nullptr;
 
-      if (scorer_flag)
+      // default behaviour uses binary
+      if (!options.was_supplied("link"))
       {
-        options.insert("link", "glf1");
+        options.insert("binary", "");
       }
       else
       {
-        options.insert("binary", "");
+        // if link was supplied then force glf1
+        if (options.get_key_value("link") != "glf1")
```

Comment:
Adding this enforces every option provider to have a string representation of the option. Instead of this you can get the typed option value, since you know this is a string.
```suggestion
        if (options.get_typed_option<std::string>("link") .value()!= "glf1")
```

Reply:
The problem with that is that `"link"` is added as an option in the `scorer` so it is not available when `cats_tree` gets set up, hence the new method `get_key_value`

Reply:
One option is to say that this reduction also requires a "link" option, and you can then do this. This is also done for CB where multiple reductions need to see the "cb_type" option